cprover_bindings/src/goto_program/symbol.rs

```diff
@@ -128,6 +129,28 @@ impl Symbol {
         )
     }
 
+    pub fn contract<T: Into<InternedString>, U: Into<InternedString>>(
+        name: T,
+        base_name: U,
+        typ: Type,
+        contract: Contract,
+        loc: Location,
+    ) -> Symbol {
+        let name = name.into();
+        // Base name is the name of the function that the contract is written for.
+        let base_name: InternedString = base_name.into();
+        let pretty_name: InternedString = base_name;
```

Comment:
Not the pretty name of the function the contract was written for?

Reply:
No. Pretty name also contains the name of the function.
Reference: [c_typecheck_base.cpp](https://github.com/diffblue/cbmc/blob/b466503080639f98af3329e9e9bb05ac847a2fc3/src/ansi-c/c_typecheck_base.cpp) (lines 880-881)